src/Language/Futhark/Parser/Lexer.x

```diff
@@ -71,6 +71,9 @@ tokens :-
   "_"                      { tokenC UNDERSCORE }
   "!"                      { tokenC BANG }
   "."                      { tokenC DOT }
+  "->"                     { tokenC TO }
+  ":"                      { tokenC HAS_TYPE }
```

Comment:
This token should be called COLON, and the one above maybe ARROW.  Tokens should have names that suggest what they look like, not what they mean (because they may be used in different contexts).